### PR TITLE
fix(consensus): make sure StreamHandler sends receiver on message zero

### PR DESCRIPTION
**Pull request by guy-starkware**
Make sure the StreamHandler doesn't send a receiver until it gets message number 0.

### CODE REVIEW

**Comment by guy-starkware** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/2748?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2748** <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/2748?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/2748?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#2746** <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/2746?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2745** <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/2745?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2730** <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/2730?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by matan-starkware — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2748)* status: all files reviewed, 1 unresolved discussion (waiting on @guy-starkware)

___
*[`crates/sequencing/papyrus_consensus/src/stream_handler.rs` line 283 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2748#-OEPOcRKC4Dy9HpT2HB3:-OEPOcRKC4Dy9HpT2HB4:undefined) ([raw file](https://github.com/starkware-libs/sequencer/blob/be4b2b6aee509bbd99547d05bd01046ff731453a/crates/sequencing/papyrus_consensus/src/stream_handler.rs#L283)):*
> ```XML
>                 // // TODO(guyn): reconsider the "expect" here.
>                 // self.inbound_channel_sender.try_send(receiver).expect("Send should succeed");
>                 StreamData::new()
> ```





_Suggestion:_
```XML
                StreamData::new()
```

**Review by guy-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2748)* status: 1 of 2 files reviewed, 1 unresolved discussion (waiting on @matan-starkware)

___
*[`crates/sequencing/papyrus_consensus/src/stream_handler.rs` line 283 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2748#-OEPOcRKC4Dy9HpT2HB3:-OEPWgkqBsAIyovo2bL3:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/be4b2b6aee509bbd99547d05bd01046ff731453a/crates/sequencing/papyrus_consensus/src/stream_handler.rs#L283)):*
> ```XML
>                 // // TODO(guyn): reconsider the "expect" here.
>                 // self.inbound_channel_sender.try_send(receiver).expect("Send should succeed");
>                 StreamData::new()
> ```

Done.

**Review by matan-starkware — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2748)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @guy-starkware)

File: crates/sequencing/papyrus_consensus/src/stream_handler_test.rs [hunk 1]
```diff
@@ -354,6 +356,9 @@ mod tests {
             stream_handler
         });
 
+        // Get the receiver for the first stream.
```
[the remark this answers is not in the document]
Comment (guy-starkware) [reply crates/sequencing/papyrus_consensus/src/stream_handler_test.rs:359]: :thumbsup:

**Review by guy-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2748)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @graphite-app[bot] and @matan-starkware)

**Review by guy-starkware — COMMENTED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2748)* status: all files reviewed, 1 unresolved discussion (waiting on @graphite-app[bot])

**Review by guy-starkware — COMMENTED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2748)* status: all files reviewed, 1 unresolved discussion (waiting on @graphite-app[bot])

**Review by guy-starkware — COMMENTED** [review]
> Dismissed @graphite-app[bot] from a discussion.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2748)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @guy-starkware)

**Comment by guy-starkware** [comment]
### Merge activity

* **Dec 23, 4:11 AM EST**: A user merged this pull request with [Graphite](https://app.graphite.dev/github/pr/starkware-libs/sequencer/2748).

**Pull request merged**